marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-utils.sjs

```diff
@@ -192,6 +209,22 @@ class HubUtils {
     return permissions;
   }
 
+  // this function can be used to create xquery/xpath templates that are safe from injection attacks
+  xquerySanitizer(strings, ...values) {
```

Comment:
I may have missed it, but this seems like an ideal candidate for some nice unit tests to verify that it's doing its thing correctly, as opposed to trying to verify this in the context of a complex integration test.

Reply:
Added a simple ml-unit-test for this.